### PR TITLE
For ubuntu-22.04 the script to delete users was not working so, versi…

### PR DESCRIPTION
**Pull request by taragurung**
1. user delete script was not working
2. added deletion script with version check, for Ubuntu:22.04 added new code to work for other version the previous code will run as it is.

Not sure if its an issue in other version but for Ubuntu:22.04 it will work

### CODE REVIEW

File: scripts/users [hunk 1]
```diff
@@ -1,9 +1,41 @@
...
+
+SCRIPT_COUNT=1
+
 function f_users {
   echo "[$SCRIPT_COUNT] Remove users"
 
-  for users in games gnats irc list news sync uucp; do
-    userdel -r "$users" 2> /dev/null
+  for user in games gnats irc list news sync uucp; do
+    sudo userdel -r "$user" 2> /dev/null
```
Comment (konstruktoid) [inline scripts/users:9]: no need for `sudo`, the script is intended to run as `root`.

File: scripts/users [hunk 2]
```diff
@@ -1,9 +1,41 @@
+#!/bin/bash
+
+SCRIPT_COUNT=1
```
Comment (konstruktoid) [inline scripts/users:3]: this shouldn't be here, `SCRIPT_COUNT` should be placed inside the functions and is started at https://github.com/konstruktoid/hardening/blob/master/scripts/pre#L4

File: scripts/users [hunk 3]
```diff
@@ -1,9 +1,41 @@
...
+    else
+      echo "User $user does not exist."
+    fi
+  done
+
+  ((SCRIPT_COUNT++))
+}
+
+# Check Ubuntu version
+if lsb_release -d | grep -q "Ubuntu 22.04"; then
```
Comment (konstruktoid) [inline scripts/users:37]: don't limit to one release.

**Review by konstruktoid — CHANGES_REQUESTED** [review]

**Comment by konstruktoid** [comment]
and correct the ShellCheck issue:
`scripts/users:23:12: note: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?. [SC2181]`
https://github.com/konstruktoid/hardening/actions/runs/9857792141/job/27239466727

File: scripts/users (hunk 3 above)
Comment (taragurung) [reply scripts/users:38]: makes sense, i was not sure how it will behave in other release so added it. but will make changes

File: scripts/users [hunk 4]
```diff
@@ -1,9 +1,42 @@
...
-    userdel -r "$users" 2> /dev/null
+  for user in games gnats irc list news sync uucp; do
+    userdel -r "$user" 2> /dev/null
   done
 
   ((SCRIPT_COUNT++))
 }
+
+function f_users_modified {
+  SCRIPT_COUNT=1
```
Comment (konstruktoid) [inline scripts/users:16]: Don't set a static `SCRIPT_COUNT`.

**Review by konstruktoid — CHANGES_REQUESTED** [review]

File: scripts/users (hunk 4 above)
Comment (taragurung) [reply scripts/users]: necessary change done and checks passed! @konstruktoid

File: scripts/users (hunk 1 above)
Comment (taragurung) [reply scripts/users]: The documentation says `sudo bash ubuntu.sh` so might be the reason why it failed for me and had to add there.

File: scripts/users [hunk 5]
```diff
@@ -1,9 +1,11 @@
+#!/bin/bash
+
 function f_users {
+  SCRIPT_COUNT=1
```
Comment (konstruktoid) [inline scripts/users:4]: This one as well

**Review by konstruktoid — CHANGES_REQUESTED** [review]

File: scripts/users (hunk 5 above)
Comment (taragurung) [reply scripts/users]: done!

**Review by konstruktoid — APPROVED** [review]

File: scripts/users [hunk 6]
```diff
@@ -1,9 +1,10 @@
+#!/bin/bash
+
 function f_users {
   echo "[$SCRIPT_COUNT] Remove users"
 
-  for users in games gnats irc list news sync uucp; do
-    userdel -r "$users" 2> /dev/null
+  for user in games gnats irc list news sync uucp; do
```
Comment (konstruktoid) [inline scripts/users:6]: Not much left of this PR, what happened to the `pkill` and test?

**Review by konstruktoid — CHANGES_REQUESTED** [review]

**Pull request closed without merging**